src/Pact/Native.hs

```diff
@@ -385,6 +387,56 @@ fromNamespacePactValue :: Namespace PactValue -> Namespace (Term Name)
 fromNamespacePactValue (Namespace n userg adming) =
   Namespace n (fromGuardPactValue userg) (fromGuardPactValue adming)
 
+dnUserGuard :: FieldKey
+dnUserGuard = "user-guard"
+
+dnAdminGuard :: FieldKey
+dnAdminGuard = "admin-guard"
+
+dnNamespaceName :: FieldKey
+dnNamespaceName = "namespace-name"
+
+describeNamespaceSchema :: NativeDef
+describeNamespaceSchema = defSchema "described-namespace"
+  "Schema type for data returned from 'describe-namespace'."
+  [ (dnUserGuard, tTyGuard Nothing)
+  , (dnAdminGuard, tTyGuard Nothing)
+  , (dnNamespaceName, tTyString)
+  ]
+
+describeNamespaceDef :: NativeDef
+describeNamespaceDef = defGasRNative "describe-namespace" describeNamespace
+  (funType (tTyObject dnTy) [("ns", tTyString)])
+  [LitExample "(describe-namespace 'my-namespace)"]
+  "Describe the namespace NS, returning a row object containing \
+  \the user and admin guards of the namespace, as well as its name."
+  where
+    dnTy = TyUser (snd describeNamespaceSchema)
+
+    describeNamespace :: GasRNativeFun e
+    describeNamespace g0 i as = case as of
+      [TLitString nsn] -> go g0 i nsn
+      _ -> argsError i as
+
+    go g0 fi n = do
+      let i = getInfo fi
+
+      mNs <- readRow i Namespaces $ NamespaceName n
+
+      case mNs of
+        Just ns@(Namespace nsn user admin) -> do
+          let guardTermOf g = TGuard (fromPactValue <$> g) def
+
+          computeGas' g0 fi (GPostRead (ReadNamespace ns)) $
+            pure $ toTObject TyAny def
+              [ (dnUserGuard, guardTermOf user)
+              , (dnAdminGuard, guardTermOf admin)
+              , (dnNamespaceName, toTerm $ renderCompactText nsn)
+              ]
+
+        Nothing -> evalError i $ "Namespace not defined: " <> pretty n
```

Comment:
I think `evalError'` allows you to just pass `fi` without having to call `getInfo`.

Reply:
`i` is already in scope!

Reply:
Stale